.github/workflows/ci.yml

```diff
@@ -502,6 +502,103 @@ jobs:
       - name: Check dirty git
         uses: ./.github/actions/check-dirty-git
 
+  mint-auditor-local-network-tests:
+    runs-on: [self-hosted, Linux, large]
+    container: mobilecoin/builder-install:v0.0.16
+
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3'
+      - name: Setup Rust
+        uses: ./.github/actions/setup-rust
+      - name: Build and generate sample data
+        run: |
+          # Generate enclave signing key
+          openssl genrsa -out Enclave_private.pem -3 3072
+          export CONSENSUS_ENCLAVE_PRIVKEY="$PWD/Enclave_private.pem"
+          export INGEST_ENCLAVE_PRIVKEY="$PWD/Enclave_private.pem"
+          export LEDGER_ENCLAVE_PRIVKEY="$PWD/Enclave_private.pem"
+          export VIEW_ENCLAVE_PRIVKEY="$PWD/Enclave_private.pem"
+          export MC_LOG=debug
+
+          # Build binaries
+          cargo build \
+              -p mc-admin-http-gateway \
+              -p mc-consensus-mint-client \
+              -p mc-consensus-service \
+              -p mc-crypto-x509-test-vectors \
+              -p mc-ledger-distribution \
+              -p mc-mint-auditor \
+              -p mc-mobilecoind \
+              -p mc-util-generate-sample-ledger \
+              -p mc-util-grpc-admin-tool \
+              -p mc-util-keyfile \
+              -p mc-util-seeded-ed25519-key-gen \
+              --release
+
+          BIN_DIR="$PWD/target/release"
+
+          # Run in temp dir to appease check-dirty-git.
+          mkdir -p /tmp/local-network
+          cd /tmp/local-network
+
+          # Generate sample keys and ledger.
+          "$BIN_DIR/sample-keys" --num 10 --seed=AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
+          "$BIN_DIR/generate-sample-ledger" --txs 100
+      - name: Run local network and mint auditor tests
+        run: |
```

Comment:
Optional: Consider moving this logic into a script, so that folks can debug CI failures by invoking that script, and we keep this config focused on invoking the steps

Reply:
I guess for the moment i want to keep it similar to the other local network test steps. I think it's okay to debug the test by running the shell chunk from ci.yml.

Maybe we should move all of this to files though, for all the tests. will think about it